### ISSUE TITLE
`@fab/server` Cache implementation assumes all `ReadableStream` objects are text

### ISSUE THREAD

**Issue by taybenlor** [fab-spec/fab#234]
In `cache.ts:readAllIfStream` it casts the `ReadableStream` to a `ReadableStream<string>` and then uses a TextEncoder to encode the data. I've been passing through images to this in a lazy way by doing `cache.set('someimage.jpg', request.body)` and it seems to be mangling the bytes when I try to retrieve them later.

I'm not sure why this is trying to force the encoding over to text, maybe you had another use case in mind?

If it should just be storing a raw Buffer, I'm happy to whip up a PR for it.

**Comment by taybenlor** [comment]
After a bit of fiddling it looks like serving using the `@fab/server` is also forcing all request bodies to be string encoded. I'm not particularly familiar with express but this middleware was causing me issues:

```
app.use((req, res, next) => {
  req.pipe(
    concat((data: any) => {
      req.body = data.toString()
      next()
    })
  )
})
```

I'm guessing you're doing this to emulate a fetch style request?

I dropped the `.toString()` and everything seemed to work fine, but there may be unintended consequences elsewhere.

**Comment by geelen** [comment]
Yeah this stuff is all a little rough, in general I just wanted to get things working for strings and was 100% going to fix it for buffers later.

If you want to whip up a PR that stores raw buffers, I'm pretty sure the caching stuff is tested well enough for strings so if the build passes I'm sure it'll be fine. At the very least, it'll be closer to where we want to get to!

As for the `req.pipe` `concat` `req.body = data.toString()` stuff, I have genuinely no idea why that has to look like that. That's just what I got working so that POST requests could have bodies, but I'm pushing my limits of understanding Node's request vs Fetch's one, so if dropping `.toString` just works then let's do that!

**Comment by taybenlor** [comment]
Yeah cool, I've got it working a bit dodge locally, I'll clean it up, run the tests and submit a PR.

**Comment by evanderkoogh** [comment]
Hey @geelen, I can image the reason for strings-only being that Buffers are a Node thing, not a JavaScript. So that if you use Buffers, it may only work in NodeJS, not V8?
